elks/arch/i86/drivers/char/serial.c

```diff
@@ -432,15 +436,19 @@ static void rs_init(void)
 	if (!rs_probe(sp)) {
 	    switch(sp->irq) {
 	    default:
-#if defined(CONFIG_NEED_IRQ4) || defined(CONFIG_NEED_IRQ3)
+#if !defined(CONFIG_FAST_IRQ3) && !defined(CONFIG_FAST_IRQ43)
 		request_irq(sp->irq, rs_irq, NULL);
 #endif
 		break;
 #ifdef CONFIG_FAST_IRQ4
 	    case 4:
+		// FIXME: how to find code segment of the fast handler ?
+		int_vector_set (0x0C, _irq_com1, /*seg_code()*/);
```

Comment:
This is getting far too messy. We don't need or want another, second way of specifying an interrupt handler for an IRQ number. Especially one where one needs to know the interrupt vector number separately from the irq number.

Proper way to handle all of this would be to use `request_irq(sp->irq, ... )` and pass a __far pointer to the interrupt handler routine; everything else is handled under the hood.

But we don't need to pass far pointer because I now realize that better method is to leave _irqit alone and keep a full copy (non-ifdef'd) of the `_irqtab` table, and have under-the-hood write that table as well as low-memory vector. This achieves any benefit of dynamic IRQs without any risk of meddling in well-tested codepaths.

Reply:
Only one way to set an IRQ handler after 1db6bc5.